libpd_wrapper/util/ringbuffer.c

```diff
@@ -14,6 +14,21 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef __APPLE__ // apple atomics
+# include <libkern/OSAtomic.h>
+# define SYNC_FETCH(ptr, val) OSAtomicOr32Barrier(val, ptr)
+# define SYNC_COMPARE(ptr, oldval, newval) \
```

Comment:
Let's call this SYNC_COMPARE_AND_SWAP since the swap is really the main part here.